### PR TITLE
feat(PL-2372): use catalog configs for release rendering

### PR DESCRIPTION
**Pull request by davidmdm**
This PR changes the rendering of joy releases to use joy's SDK rendering which takes into account catalog configurations such as:
- valueMapping(s)
- defaultChart

The goal of these changes is to be able to remove nesto specific configuration from the appset and have it be directly configurable via the joy catalog.

## Default Chart from catalog being injected
image

## Annotation being mapped in from valueMapping
image

## Image tag being mapped in from valueMapping
image

### CODE REVIEW

**Review by riclima — APPROVED** [review]

**Review by riclima — APPROVED** [review]

**Review by ayahajar — APPROVED** [review]

File: internal/apiserver/apiserver.go [hunk 1]
```diff
@@ -31,38 +32,38 @@ type GetParamsRequest struct {
...
 	return &ApiServer{
 		token:     token,
 		generator: g,
+		repo:      repo,
 	}
 }
 
-func (s *ApiServer) Run() error {
-	r := gin.New()
+func (server *ApiServer) Run() error {
```
Comment (alexstojda) [inline internal/apiserver/apiserver.go:46]: By convention, receiver names should generally by 1-2 characters. Makes receiver variables easy to differentiate from other variables. 

You also change it here, but not in other functions. So lets just keep it as `s`

From go.dev: https://go.dev/wiki/CodeReviewComments#receiver-names

File: internal/apiserver/apiserver.go [hunk 2]
```diff
@@ -71,13 +72,11 @@ func (s *ApiServer) Readiness(c *gin.Context) {
...
 
-	c.JSON(http.StatusOK, gin.H{
-		"status": "ok",
-	})
+	c.JSON(http.StatusOK, gin.H{"status": "ok"})
 }
 
-func (s *ApiServer) GetParamsExecute(c *gin.Context) {
-	if c.GetHeader("Authorization") != "Bearer "+s.token {
+func (server *ApiServer) GetParamsExecute(c *gin.Context) {
```
Comment (alexstojda) [inline internal/apiserver/apiserver.go:78]: See receiver naming comment above

File: internal/generator/generator.go [hunk 3]
```diff
@@ -25,58 +29,109 @@ type Result struct {
...
+		if err != nil {
+			return nil, fmt.Errorf("loading catalog: %w", err)
+		}
+
+		cfg, err := joy.LoadCatalogConfig(repo.Directory())
+		if err != nil {
+			return nil, fmt.Errorf("loading catalog config: %w", err)
+		}
+
+		return &JoyContext{Catalog: cat, Config: cfg}, nil
```
Comment (alexstojda) [inline internal/generator/generator.go:60]: Is this extraction to make the generator code testable?

**Review by alexstojda — CHANGES_REQUESTED** [review]
Looks good so far, just some minor things & a question.

Please also add some comments for the new code since it was not clear at first glance why you extracted some code into a different method, as well as what the purpose of that code is.

**Review by alexstojda — APPROVED** [review]

**Pull request merged**